src/index.js

```diff
@@ -679,36 +679,45 @@ class NearProvider {
         try {
             let outcome;
             let val;
-            const { to, value, data } = txObj;
+            const { from, to, value, data } = txObj;
             if (value === undefined) {
                 val = new BN(0)
             } else {
                 const remove = utils.remove0x(value)
                 val = new BN(remove, 16)
             }
-            // value === undefined
-            //     ? new BN(0)
-            //     : new BN(utils.remove0x(value), 16);
-
-            // TODO: differentiate simple sends
-            const functionCallData = {
-                contractId: this.evm_contract,
-                methodName: 'deploy_code',
-                args: { 'bytecode': utils.remove0x(data) },
-                gas: GAS_AMOUNT,
-                amount: val
-            }
 
             if (to === undefined) {
-                // Contract deployment.
+                // Contract deployment
                 outcome = await this.account.functionCall(
                     this.evm_contract,
                     'deploy_code',
                     { bytecode: utils.remove0x(data) },
                     GAS_AMOUNT,
                     val
                 );
+            } else if (to != ZERO_ADDRESS && to == from) {
```

Comment:
wingin' it here, but thought this made sense!

Reply:
let me check and see if I have this right:
```
to === undefined -> deploy contract
to === from -> add_near (move near from near account to EVM account)
data === undefined -> move_funds_to_evm_address
 else -> call_contract
```

I think that's pretty reasonable. Self-sends fund the near EVM account. Sends to others work as expected

Reply:
correct. I don't LOVE the current state of the conditionals and the fact that the order matters...could make them more verbose but a little sturdier.

Reply:
also this doesn't account for: `retrieve_near`  & `move_funds_to_near_account` yet